test-framework/common/src/main/java/io/quarkus/test/common/http/TestHTTPResourceManager.java

```diff
@@ -12,7 +12,13 @@
 public class TestHTTPResourceManager {
 
     public static String getUri() {
-        return ConfigProvider.getConfig().getValue("test.url", String.class);
+        try {
+            return ConfigProvider.getConfig().getValue("test.url", String.class);
+        } catch (IllegalStateException e) {
+            //massive hack for dev mode tests, dev mode has not started yet
+            //so we don't have any way to load this correctly from config
+            return "http://localhost:8080";
```

Comment:
Isn't the test port usually 8081?

Reply:
Dev mode tests run in mode DEVELOPMENT not TEST, so the port is 8080